contract/src/uri.ts

```diff
@@ -38,9 +39,8 @@ export class Uri implements ParsedUri {
   /** {@link https://tools.ietf.org/html/rfc3986#section-5.3} */
   toString() {
     const result: string[] = [];
-
     if (typeof this.scheme != 'undefined') result.push(this.scheme, ":");
-    if (typeof this.authority != 'undefined') result.push("//", this.authority);
+    if (typeof this.authority != 'undefined' && typeof this.authority.host != 'undefined') result.push("//", this.authority.toString());
```

Comment:
You shouldn't need to do this: typeof this.authority.host != 'undefined'

Your breaking encapsulation